AutoGrade/grader.py

```diff
@@ -116,4 +90,6 @@ def run_student_tests(target_folder, total_points, timeout):
     logger.debug("Read test line [" + res_line.strip("=") + "]")
     logger.debug("Calculated score: " + str(score))
 
-    return [score, out]
+    timeout = (out.find("+ Timeout +") != -1)
```

Comment:
When a single test case fail then the whole execution is stopped (for my PC =]) and such error is generated.
 
"""
...
+++++++++++++ Timeout +++++++++++++
"""

That is why using "+ Timeout +" to detect the timeout.

Reply:
Do you happen to get this behavior on Windows or are you running Linux?

Reply:
I am getting this behavior on Windows... just check in "Bash on Ubuntu" and it is working same as you stated (1 passes and 1 failed).

Reply:
OK, good. Forget about Windows. Let's just only remove the `timeout` from the join call. Everything else can remain the same (don't want to experiment with `popen` right now.) 

Thanks for looking into this.

Reply:
Using Popen because log was not generating properly in run_test method.

Reply:
Why not? It's been working fine till now Oo

Reply:
Its behavior was different when "+ Timeout +" jumps in =]

Reply:
Yes but we're not expecting that on prod =] 

Keep your diff safe but let's commit with only the `timeout` removed from the `join` at the moment. 

(p.s. Autograder is down probably because of electricity issues at the campus ... )

Reply:
Ok =]

Reply:
Thanks. 

Oh and let's squash and merge this branch so that we don't have all the extra stuff in the commit on master.

Reply:
Keeping this line for my environment =]

Reply:
Also need to revert the timeout bits `timeout = (out.find("+ Timeout +") != -1)` ... and the changes in `views.py`.

Reply:
I have kept that line... so that AutoGrader doesn't break on my machine... BTW that is just one line =]

Reply:
Ok, you can keep the line =]  ... Just put a comment in there that mentions this whole conversation (for future reference about why it's there.)